tracer.go

```diff
@@ -187,15 +203,17 @@ func (t *RemoteTracer) doWrite() {
 	for {
 		_, ok := <-t.ch
 
-		// nil out the buffer to gc events when swapping buffers
-		for i := range buf {
-			buf[i] = nil
-		}
-
-		// wait a bit to accumulate a batch
-		time.Sleep(time.Second)
+		// deadline for batch accumulation
+		deadline := time.Now().Add(time.Second)
 
+	again:
 		t.mx.Lock()
+		if len(t.buf) < MinTraceBatchSize && time.Now().Before(deadline) {
+			t.mx.Unlock()
+			time.Sleep(100 * time.Millisecond)
+			goto again
+		}
```

Comment:
for loop?

```go
t.mx.Lock()
for len(t.buf) < MinTraceBatchSize && time.Now().Before(deadline) { 
  t.mx.Unlock()
  time.Sleep(100 * time.Millisecond)
  t.mx.Lock()
}
```

Reply:
the alergy to gotos strikes again! Ok, I'll write it as a for loop.